src/bootstrap/compile.rs

```diff
@@ -640,7 +642,9 @@ impl Step for CodegenBackend {
             .arg(builder.src.join("src/librustc_codegen_llvm/Cargo.toml"));
         rustc_cargo_env(builder, &mut cargo);
 
-        features += &build_codegen_backend(&builder, &mut cargo, &compiler, target, backend);
+        if !backend.is_empty() {
```

Comment:
This is probably not the check you wanted -- we shouldn't be creating `CodegenBackend` with an empty string. I think we can probably just remove this if.

Reply:
What exactly do you mean?

Reply:
`backend` here isn't an array, but a `Interned<String>` -- this check should be removed.